pkg/innerring/innerring.go

```diff
@@ -376,21 +352,57 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper) (*Server, error
 			return nil, err
 		}
 
-		// enable notary support in the client
-		var mainnetNotaryOpts []client.NotaryOption
-		if !server.mainNotaryConfig.disabled {
-			mainnetNotaryOpts = append(mainnetNotaryOpts,
-				client.WithProxyContract(server.contracts.processing),
-				client.WithAlphabetSource(server.morphClient.Committee))
-		}
-
 		// create mainnet client
-		server.mainnetClient, err = createClient(ctx, mainnetChain, mainnetNotaryOpts...)
+		server.mainnetClient, err = createClient(ctx, mainnetChain)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	server.mainNotaryConfig, server.sideNotaryConfig = parseNotaryConfigs(
+		cfg,
+		server.morphClient.ProbeNotary(),
+		!withoutMainNet && server.mainnetClient.ProbeNotary(),
```

Comment:
Isn't it will be the same thing in the end?
```suggestion
		server.mainnetClient.ProbeNotary(),
```

Reply:
if i remember correctly what i wanted to do:

if `withoutMainNet == true` and notary is enabled, then `server.mainnetClient` is `server.morphClient` in fact, then `ProbeNotary()` is called on sidechain in fact, then notary enabled in sidechain causes IR to require notary main contracts. it is not correct

Reply:
> then notary enabled in sidechain causes IR to require notary main contracts. it is not correct

You mean `EnableNotarySupport()` will be called twice on morph chain client? If so, we can make `EnableNotarySupport()` to ignore if it was already enabled.

However I am fine with this solution if you add some comment about that. Looks a bit confusing at first glance.

Reply:
no-no, i mean, that the line that you suggest to change is arg to `parseNotaryConfigs` func

that arg defines if notary is presented in main and if it is necessary to parse additional envs

if use your variant in "`withoutMain==true`, notary in sidechain enabled" case, that arg is true since `server.mainnetClient.ProbeNotary()` is sidechain in fact => `server.mainNotaryConfig.disabled` is false => IR try to parse additional contracts hashes because of the line `372` and try to enable it because of the line `389`

Reply:
added commentary to that argument